packages/@glimmer/syntax/lib/generation/print.ts

```diff
@@ -57,7 +57,7 @@ export default function build(ast) {
     }
     break;
     case 'MustacheCommentStatement': {
-      output.push(compactJoin(['{{!', ast.value, '}}']));
+      output.push(compactJoin(['{{!--', ast.value, '--}}']));
```

Comment:
It should be straight forward to notate the type in the AST then use it here (which both fixes the issue you were trying to fix and makes the AST more accurate).

Ping me in slack and I can explain how to do it...